docs/api/navigation-history.md

```diff
@@ -74,3 +74,15 @@ Returns `boolean` - Whether the navigation entry was removed from the webContent
 #### `navigationHistory.getAllEntries()`
 
 Returns [`NavigationEntry[]`](structures/navigation-entry.md) - WebContents complete history.
+
+#### `navigationHistory.restore(index, entries)`
+
+Restores navigation history. Will make a best effort to restore not just the navigation stack but
+also the state of the individual pages - for instance including HTML form values or the scroll
+position. It's recommended to call this API before any navigation entries are created, so ideally
```

Comment:
> It's recommended to call this API before any navigation entries are created, so ideally

Does that mean you shouldn't call this while your app is running? E.g. `foo.restore()` is for use _only_ when making a new webContents? Maybe it should be part of webContents construction logic instead then?

Reply:
We talked in person and decided:
- to throw if there's a `NavigationEntry` that's not the initial entry (aka you loaded anything)
- to re-use the loadURL logic to make restore await-able and include the full load